### DISCUSSION TITLE
guides/optimizing-openvino-latency-vs-throughput-modes/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#13602] [Docs]
# guides/optimizing-openvino-latency-vs-throughput-modes/

Discover how to enhance Ultralytics YOLO model performance using Intel's OpenVINO toolkit. Boost latency and throughput efficiently.

https://docs.ultralytics.com/guides/optimizing-openvino-latency-vs-throughput-modes/

**Comment by Kvsoni** [comment]
Is it possible to use the in built classes and functions of ultralytics to do pre-processing and post-processing and using openVINO for the inference part?

**Reply by pderrenger** [reply]
@Kvsoni hi there!

Yes, it is indeed possible to use Ultralytics' built-in classes and functions for pre-processing and post-processing while leveraging OpenVINO for the inference part. Here's a high-level overview of how you can achieve this:

1. **Pre-processing with Ultralytics**: You can use the pre-processing utilities provided by Ultralytics to prepare your input data. This includes resizing, normalization, and other transformations.

2. **Inference with OpenVINO**: Once your data is pre-processed, you can pass it to an OpenVINO-optimized model for inference. OpenVINO provides significant performance improvements, especially for Intel hardware.

3. **Post-processing with Ultralytics**: After inference, you can use Ultralytics' post-processing functions to interpret the results, such as applying non-maximum suppression (NMS) and decoding the output.

Here’s a simplified example to illustrate this workflow:

```python
from ultralytics import YOLO
import openvino.runtime as ov
import numpy as np

# Load Ultralytics model for pre-processing and post-processing
model = YOLO('yolov8n.pt')

# Pre-process the input image
img = model.preprocess('path/to/image.jpg')

# Load OpenVINO model for inference
core = ov.Core()
compiled_model = core.compile_model('path/to/openvino_model.xml', 'CPU')

# Perform inference
input_tensor = ov.Tensor(array=img, shape=(1, 3, 640, 640))
results = compiled_model([input_tensor])

# Post-process the results
output = model.postprocess(results)

# Display or save the results
model.show(output)
```

For more detailed guidance on optimizing your model with OpenVINO, including strategies for latency and throughput, please refer to our comprehensive guide: [Optimizing OpenVINO Inference for Ultralytics YOLO Models](https://docs.ultralytics.com/guides/optimizing-openvino-latency-vs-throughput-modes/).

Feel free to reach out if you have any more questions or need further assistance! 😊
